[Frontend] Added support for HermesToolParser for models without special tokens

minpeter opened the pull request:
Upstream the implementation implemented in the https://github.com/minpeter/hermes-llama-parse repo,

Modify the hermes parser to be compatible with models that do not have separate "<tool_call>", "</tool_call>" tags (e.g. llama),

*I know there is a llama-specific parser, but it is useful in fine-tuning llama-based models.

minpeter committed:
Refactor HermesToolParser for improved readability and functionality

Signed-off-by: minpeter <kali2005611@gmail.com>

minpeter commented:
@mgoin Is there a review schedule or deadline that I can expect?

minpeter committed:
Merge branch 'vllm-project:main' into hermes-parse-improve

minpeter committed:
Merge branch 'vllm-project:main' into hermes-parse-improve

minpeter committed:
Merge branch 'vllm-project:main' into hermes-parse-improve

minpeter committed:
Merge branch 'vllm-project:main' into hermes-parse-improve

aarnphm reviewed (commented):
Is there a test fine-tuned model that we can use to test this?

maybe a very small finetune llama3.2-1b would work here.

minpeter commented:
> Is there a test fine-tuned model that we can use to test this?
> 
> maybe a very small finetune llama3.2-1b would work here.

https://huggingface.co/minpeter/LoRA-Llama-3b-v1-iteration-00-sf-xlam-10

minpeter committed:
style: snake_case Naming Convention

According to the project's naming convention,
'Buffered_delta_text' will be changed to 'buffered_delta_text'.
This will improve the consistency and readability of the code.

Signed-off-by: minpeter <kali2005611@gmail.com>

aarnphm commented:
Can you add a testcase using that model?

aarnphm commented:
iirc there are a few tool parser tests under `tests/entrypoints/openai/tool_parsers`

minpeter committed:
Merge branch 'vllm-project:main' into hermes-parse-improve

minpeter committed:
drift test code for Hermes-2-Pro tool parser functionality

Signed-off-by: minpeter <kali2005611@gmail.com>

minpeter committed:
fix lint

Signed-off-by: minpeter <kali2005611@gmail.com>

minpeter committed:
refactor: streamline Hermes tool parser tests and enhance functionality

Signed-off-by: minpeter <kali2005611@gmail.com>

minpeter committed:
model update, 3b -> 1b

Signed-off-by: minpeter <kali2005611@gmail.com>

minpeter committed:
Merge branch 'vllm-project:main' into hermes-parse-improve

minpeter committed:
Merge branch 'vllm-project:main' into hermes-parse-improve

minpeter committed:
Merge branch 'main' into hermes-parse-improve
